Rombie's change to Stella (Bro Code Technologies, 2026-10-29), under Stella's GPL-2.0-or-later, for Stella 7.0c
(c4016ea): the core is built without the code Stella says it took from z26, whose license (the GPL "Version 2",
with no "or any later version") cannot be combined with RetroArch's GPL-3.0. Removed:
- the CompuMate (cartridge scheme CM and its keyboard): a CompuMate cartridge no longer loads;
- the Kid Vid, the MindLink, the Amiga mouse, the Atari mouse and the trackball: a game set to one of them gets a
  joystick;
- the mouse driving a joystick, and the trackball sensitivity setting;
- the z26 palette: choosing it gives Stella's standard palette, and the libretro option no longer offers it;
- the default Supercharger header: a 6K image now gets one written from the load format's fields.
stella/build.sh also deletes the removed classes' files (CartCM, CompuMate, KidVid, MindLink, PointingDevice,
AmigaMouse, AtariMouse, TrakBall) from the build's copy, so nothing compiled can include them.

diff --git a/src/common/PJoystickHandler.cxx b/src/common/PJoystickHandler.cxx
index d21bdef..f00829f 100644
--- a/src/common/PJoystickHandler.cxx
+++ b/src/common/PJoystickHandler.cxx
@@ -18,7 +18,6 @@
 #include "OSystem.hxx"
 #include "Console.hxx"
 #include "Paddles.hxx"
-#include "PointingDevice.hxx"
 #include "Driving.hxx"
 #include "Settings.hxx"
 #include "EventHandler.hxx"
@@ -1277,21 +1276,8 @@ void PhysicalJoystickHandler::changeMousePaddleSensitivity(int direction)
 }
 
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-void PhysicalJoystickHandler::changeMouseTrackballSensitivity(int direction)
+void PhysicalJoystickHandler::changeMouseTrackballSensitivity(int)
 {
-  const int sense =
-    BSPF::clamp(myOSystem.settings().getInt("tsense") + direction,
-                PointingDevice::MIN_SENSE, PointingDevice::MAX_SENSE);
-  myOSystem.settings().setValue("tsense", sense);
-
-  PointingDevice::setSensitivity(sense);
-
-  ostringstream ss;
-  ss << sense * 10 << "%";
-
-  myOSystem.frameBuffer().showGaugeMessage(
-    "Mouse trackball sensitivity", ss.view(), sense,
-    PointingDevice::MIN_SENSE, PointingDevice::MAX_SENSE);
 }
 
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
diff --git a/src/common/PaletteHandler.cxx b/src/common/PaletteHandler.cxx
index 6e39ce5..ce29348 100644
--- a/src/common/PaletteHandler.cxx
+++ b/src/common/PaletteHandler.cxx
@@ -325,7 +325,7 @@ void PaletteHandler::setPalette()
     static constexpr BSPF::array2D<const PaletteArray*, PaletteType::NumTypes,
     static_cast<int>(ConsoleTiming::numTimings)> palettes = {{
       { &ourNTSCPalette,       &ourPALPalette,       &ourSECAMPalette     },
-      { &ourNTSCPaletteZ26,    &ourPALPaletteZ26,    &ourSECAMPaletteZ26  },
+      { &ourNTSCPalette,       &ourPALPalette,       &ourSECAMPalette     },
       { &ourUserNTSCPalette,   &ourUserPALPalette,   &ourUserSECAMPalette },
       { &ourCustomNTSCPalette, &ourCustomPALPalette, &ourSECAMPalette     }
     }};
@@ -739,114 +739,6 @@ const PaletteArray PaletteHandler::ourSECAMPalette = {
   0x7fff00, 0, 0x7fffff, 0, 0xffff3f, 0, 0xffffff, 0
 };
 
-// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-const PaletteArray PaletteHandler::ourNTSCPaletteZ26 = {
-  0x000000, 0, 0x505050, 0, 0x646464, 0, 0x787878, 0,
-  0x8c8c8c, 0, 0xa0a0a0, 0, 0xb4b4b4, 0, 0xc8c8c8, 0,
-  0x445400, 0, 0x586800, 0, 0x6c7c00, 0, 0x809000, 0,
-  0x94a414, 0, 0xa8b828, 0, 0xbccc3c, 0, 0xd0e050, 0,
-  0x673900, 0, 0x7b4d00, 0, 0x8f6100, 0, 0xa37513, 0,
-  0xb78927, 0, 0xcb9d3b, 0, 0xdfb14f, 0, 0xf3c563, 0,
-  0x7b2504, 0, 0x8f3918, 0, 0xa34d2c, 0, 0xb76140, 0,
-  0xcb7554, 0, 0xdf8968, 0, 0xf39d7c, 0, 0xffb190, 0,
-  0x7d122c, 0, 0x912640, 0, 0xa53a54, 0, 0xb94e68, 0,
-  0xcd627c, 0, 0xe17690, 0, 0xf58aa4, 0, 0xff9eb8, 0,
-  0x730871, 0, 0x871c85, 0, 0x9b3099, 0, 0xaf44ad, 0,
-  0xc358c1, 0, 0xd76cd5, 0, 0xeb80e9, 0, 0xff94fd, 0,
-  0x5d0b92, 0, 0x711fa6, 0, 0x8533ba, 0, 0x9947ce, 0,
-  0xad5be2, 0, 0xc16ff6, 0, 0xd583ff, 0, 0xe997ff, 0,
-  0x401599, 0, 0x5429ad, 0, 0x683dc1, 0, 0x7c51d5, 0,
-  0x9065e9, 0, 0xa479fd, 0, 0xb88dff, 0, 0xcca1ff, 0,
-  0x252593, 0, 0x3939a7, 0, 0x4d4dbb, 0, 0x6161cf, 0,
-  0x7575e3, 0, 0x8989f7, 0, 0x9d9dff, 0, 0xb1b1ff, 0,
-  0x0f3480, 0, 0x234894, 0, 0x375ca8, 0, 0x4b70bc, 0,
-  0x5f84d0, 0, 0x7398e4, 0, 0x87acf8, 0, 0x9bc0ff, 0,
-  0x04425a, 0, 0x18566e, 0, 0x2c6a82, 0, 0x407e96, 0,
-  0x5492aa, 0, 0x68a6be, 0, 0x7cbad2, 0, 0x90cee6, 0,
-  0x044f30, 0, 0x186344, 0, 0x2c7758, 0, 0x408b6c, 0,
-  0x549f80, 0, 0x68b394, 0, 0x7cc7a8, 0, 0x90dbbc, 0,
-  0x0f550a, 0, 0x23691e, 0, 0x377d32, 0, 0x4b9146, 0,
-  0x5fa55a, 0, 0x73b96e, 0, 0x87cd82, 0, 0x9be196, 0,
-  0x1f5100, 0, 0x336505, 0, 0x477919, 0, 0x5b8d2d, 0,
-  0x6fa141, 0, 0x83b555, 0, 0x97c969, 0, 0xabdd7d, 0,
-  0x344600, 0, 0x485a00, 0, 0x5c6e14, 0, 0x708228, 0,
-  0x84963c, 0, 0x98aa50, 0, 0xacbe64, 0, 0xc0d278, 0,
-  0x463e00, 0, 0x5a5205, 0, 0x6e6619, 0, 0x827a2d, 0,
-  0x968e41, 0, 0xaaa255, 0, 0xbeb669, 0, 0xd2ca7d, 0
-};
-
-// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-const PaletteArray PaletteHandler::ourPALPaletteZ26 = {
-  0x000000, 0, 0x4c4c4c, 0, 0x606060, 0, 0x747474, 0,
-  0x888888, 0, 0x9c9c9c, 0, 0xb0b0b0, 0, 0xc4c4c4, 0,
-  0x000000, 0, 0x4c4c4c, 0, 0x606060, 0, 0x747474, 0,
-  0x888888, 0, 0x9c9c9c, 0, 0xb0b0b0, 0, 0xc4c4c4, 0,
-  0x533a00, 0, 0x674e00, 0, 0x7b6203, 0, 0x8f7617, 0,
-  0xa38a2b, 0, 0xb79e3f, 0, 0xcbb253, 0, 0xdfc667, 0,
-  0x1b5800, 0, 0x2f6c00, 0, 0x438001, 0, 0x579415, 0,
-  0x6ba829, 0, 0x7fbc3d, 0, 0x93d051, 0, 0xa7e465, 0,
-  0x6a2900, 0, 0x7e3d12, 0, 0x925126, 0, 0xa6653a, 0,
-  0xba794e, 0, 0xce8d62, 0, 0xe2a176, 0, 0xf6b58a, 0,
-  0x075b00, 0, 0x1b6f11, 0, 0x2f8325, 0, 0x439739, 0,
-  0x57ab4d, 0, 0x6bbf61, 0, 0x7fd375, 0, 0x93e789, 0,
-  0x741b2f, 0, 0x882f43, 0, 0x9c4357, 0, 0xb0576b, 0,
-  0xc46b7f, 0, 0xd87f93, 0, 0xec93a7, 0, 0xffa7bb, 0,
-  0x00572e, 0, 0x106b42, 0, 0x247f56, 0, 0x38936a, 0,
-  0x4ca77e, 0, 0x60bb92, 0, 0x74cfa6, 0, 0x88e3ba, 0,
-  0x6d165f, 0, 0x812a73, 0, 0x953e87, 0, 0xa9529b, 0,
-  0xbd66af, 0, 0xd17ac3, 0, 0xe58ed7, 0, 0xf9a2eb, 0,
-  0x014c5e, 0, 0x156072, 0, 0x297486, 0, 0x3d889a, 0,
-  0x519cae, 0, 0x65b0c2, 0, 0x79c4d6, 0, 0x8dd8ea, 0,
-  0x5f1588, 0, 0x73299c, 0, 0x873db0, 0, 0x9b51c4, 0,
-  0xaf65d8, 0, 0xc379ec, 0, 0xd78dff, 0, 0xeba1ff, 0,
-  0x123b87, 0, 0x264f9b, 0, 0x3a63af, 0, 0x4e77c3, 0,
-  0x628bd7, 0, 0x769feb, 0, 0x8ab3ff, 0, 0x9ec7ff, 0,
-  0x451e9d, 0, 0x5932b1, 0, 0x6d46c5, 0, 0x815ad9, 0,
-  0x956eed, 0, 0xa982ff, 0, 0xbd96ff, 0, 0xd1aaff, 0,
-  0x2a2b9e, 0, 0x3e3fb2, 0, 0x5253c6, 0, 0x6667da, 0,
-  0x7a7bee, 0, 0x8e8fff, 0, 0xa2a3ff, 0, 0xb6b7ff, 0,
-  0x000000, 0, 0x4c4c4c, 0, 0x606060, 0, 0x747474, 0,
-  0x888888, 0, 0x9c9c9c, 0, 0xb0b0b0, 0, 0xc4c4c4, 0,
-  0x000000, 0, 0x4c4c4c, 0, 0x606060, 0, 0x747474, 0,
-  0x888888, 0, 0x9c9c9c, 0, 0xb0b0b0, 0, 0xc4c4c4, 0
-};
-
-// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-const PaletteArray PaletteHandler::ourSECAMPaletteZ26 = {
-  0x000000, 0, 0x2121ff, 0, 0xf03c79, 0, 0xff3cff, 0,
-  0x7fff00, 0, 0x7fffff, 0, 0xffff3f, 0, 0xffffff, 0,
-  0x000000, 0, 0x2121ff, 0, 0xf03c79, 0, 0xff3cff, 0,
-  0x7fff00, 0, 0x7fffff, 0, 0xffff3f, 0, 0xffffff, 0,
-  0x000000, 0, 0x2121ff, 0, 0xf03c79, 0, 0xff3cff, 0,
-  0x7fff00, 0, 0x7fffff, 0, 0xffff3f, 0, 0xffffff, 0,
-  0x000000, 0, 0x2121ff, 0, 0xf03c79, 0, 0xff3cff, 0,
-  0x7fff00, 0, 0x7fffff, 0, 0xffff3f, 0, 0xffffff, 0,
-  0x000000, 0, 0x2121ff, 0, 0xf03c79, 0, 0xff3cff, 0,
-  0x7fff00, 0, 0x7fffff, 0, 0xffff3f, 0, 0xffffff, 0,
-  0x000000, 0, 0x2121ff, 0, 0xf03c79, 0, 0xff3cff, 0,
-  0x7fff00, 0, 0x7fffff, 0, 0xffff3f, 0, 0xffffff, 0,
-  0x000000, 0, 0x2121ff, 0, 0xf03c79, 0, 0xff3cff, 0,
-  0x7fff00, 0, 0x7fffff, 0, 0xffff3f, 0, 0xffffff, 0,
-  0x000000, 0, 0x2121ff, 0, 0xf03c79, 0, 0xff3cff, 0,
-  0x7fff00, 0, 0x7fffff, 0, 0xffff3f, 0, 0xffffff, 0,
-  0x000000, 0, 0x2121ff, 0, 0xf03c79, 0, 0xff3cff, 0,
-  0x7fff00, 0, 0x7fffff, 0, 0xffff3f, 0, 0xffffff, 0,
-  0x000000, 0, 0x2121ff, 0, 0xf03c79, 0, 0xff3cff, 0,
-  0x7fff00, 0, 0x7fffff, 0, 0xffff3f, 0, 0xffffff, 0,
-  0x000000, 0, 0x2121ff, 0, 0xf03c79, 0, 0xff3cff, 0,
-  0x7fff00, 0, 0x7fffff, 0, 0xffff3f, 0, 0xffffff, 0,
-  0x000000, 0, 0x2121ff, 0, 0xf03c79, 0, 0xff3cff, 0,
-  0x7fff00, 0, 0x7fffff, 0, 0xffff3f, 0, 0xffffff, 0,
-  0x000000, 0, 0x2121ff, 0, 0xf03c79, 0, 0xff3cff, 0,
-  0x7fff00, 0, 0x7fffff, 0, 0xffff3f, 0, 0xffffff, 0,
-  0x000000, 0, 0x2121ff, 0, 0xf03c79, 0, 0xff3cff, 0,
-  0x7fff00, 0, 0x7fffff, 0, 0xffff3f, 0, 0xffffff, 0,
-  0x000000, 0, 0x2121ff, 0, 0xf03c79, 0, 0xff3cff, 0,
-  0x7fff00, 0, 0x7fffff, 0, 0xffff3f, 0, 0xffffff, 0,
-  0x000000, 0, 0x2121ff, 0, 0xf03c79, 0, 0xff3cff, 0,
-  0x7fff00, 0, 0x7fffff, 0, 0xffff3f, 0, 0xffffff, 0
-};
-
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 PaletteArray PaletteHandler::ourUserNTSCPalette  = { 0 }; // filled from external file
 
diff --git a/src/common/PaletteHandler.hxx b/src/common/PaletteHandler.hxx
index 980958a..c0d5fa8 100644
--- a/src/common/PaletteHandler.hxx
+++ b/src/common/PaletteHandler.hxx
@@ -309,10 +309,6 @@ class PaletteHandler
     static const PaletteArray ourPALPalette;
     static const PaletteArray ourSECAMPalette;
 
-    // Table of RGB values for NTSC, PAL and SECAM - Z26 version
-    static const PaletteArray ourNTSCPaletteZ26;
-    static const PaletteArray ourPALPaletteZ26;
-    static const PaletteArray ourSECAMPaletteZ26;
 
     // Table of RGB values for NTSC, PAL and SECAM - user-defined
     static PaletteArray ourUserNTSCPalette;
diff --git a/src/emucore/CartAR.cxx b/src/emucore/CartAR.cxx
index 8fc1527..6914ce5 100644
--- a/src/emucore/CartAR.cxx
+++ b/src/emucore/CartAR.cxx
@@ -40,10 +40,27 @@ CartridgeAR::CartridgeAR(const ByteBuffer& image, size_t size,
   myLoadImages = make_unique<uInt8[]>(mySize);
   std::copy_n(image.get(), size, myLoadImages.get());
 
-  // Add header if image doesn't include it
+  // A 6K image has no load header, so give it one from the load format:
+  // start at $FAAC with bank configuration $0F, load 0, and the 24 pages in
+  // order into banks 0 to 2. The first 8 bytes sum to $55, as the loader checks.
   if(size < LOAD_SIZE)
-    std::copy_n(ourDefaultHeader.data(), ourDefaultHeader.size(),
-                myLoadImages.get() + myImage.size());
+  {
+    uInt8* header = myLoadImages.get() + myImage.size();
+    std::fill_n(header, 256, uInt8{0xff});
+    header[0] = 0xac;  header[1] = 0xfa;
+    header[2] = 0x0f;
+    header[3] = 24;
+    header[4] = 0;
+    header[5] = 0;
+    header[6] = 0x24;  header[7] = 0x02;
+    header[4] = static_cast<uInt8>(0x55 - checksum(header, 8));
+    for(uInt8 j = 0; j < 24; ++j)
+    {
+      header[16 + j] = static_cast<uInt8>(((j % 8) << 2) | (j / 8));
+      header[64 + j] = 0;
+    }
+    header[255] = 0;
+  }
 
   // We use System::PageAccess.romAccessBase, but don't allow its use
   // through a pointer, since the AR scheme doesn't support bankswitching
diff --git a/src/emucore/CartAR.hxx b/src/emucore/CartAR.hxx
index 97ea6bc..33720b4 100644
--- a/src/emucore/CartAR.hxx
+++ b/src/emucore/CartAR.hxx
@@ -233,42 +233,6 @@ class CartridgeAR : public Cartridge
     // This is not marked 'constexpr', since it's patched at runtime
     static std::array<uInt8, 294> ourDummyROMCode;
 
-    // Default 256-byte header to use if one isn't included in the ROM
-    // This data comes from z26
-    static constexpr std::array<uInt8, 256> ourDefaultHeader = {
-      0xac, 0xfa, 0x0f, 0x18, 0x62, 0x00, 0x24, 0x02,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0x00, 0x04, 0x08, 0x0c, 0x10, 0x14, 0x18, 0x1c,
-      0x01, 0x05, 0x09, 0x0d, 0x11, 0x15, 0x19, 0x1d,
-      0x02, 0x06, 0x0a, 0x0e, 0x12, 0x16, 0x1a, 0x1e,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x00
-    };
 
   private:
     // Following constructors and assignment operators not supported
diff --git a/src/emucore/CartCreator.cxx b/src/emucore/CartCreator.cxx
index 508776f..4e1ce63 100644
--- a/src/emucore/CartCreator.cxx
+++ b/src/emucore/CartCreator.cxx
@@ -33,7 +33,6 @@
 #include "CartBFSC.hxx"
 #include "CartBUS.hxx"
 #include "CartCDF.hxx"
-#include "CartCM.hxx"
 #include "CartCTY.hxx"
 #include "CartCV.hxx"
 #include "CartDF.hxx"
@@ -252,7 +251,7 @@ CartCreator::createFromImage(const ByteBuffer& image, size_t size,
     case _BFSC: return make_unique<CartridgeBFSC>(image, size, md5, settings);
     case _BUS:  return make_unique<CartridgeBUS>(image, size, md5, settings);
     case _CDF:  return make_unique<CartridgeCDF>(image, size, md5, settings);
-    case _CM:   return make_unique<CartridgeCM>(image, size, md5, settings);
+    case _CM:   throw runtime_error("CompuMate cartridges are not supported in this build");
     case _CTY:  return make_unique<CartridgeCTY>(image, size, md5, settings);
     case _CV:   return make_unique<CartridgeCV>(image, size, md5, settings);
     case _DF:   return make_unique<CartridgeDF>(image, size, md5, settings);
diff --git a/src/emucore/Console.cxx b/src/emucore/Console.cxx
index 55dbc9c..e5769e7 100644
--- a/src/emucore/Console.cxx
+++ b/src/emucore/Console.cxx
@@ -21,19 +21,12 @@
 #include "Booster.hxx"
 #include "Cart.hxx"
 #include "Control.hxx"
-#include "CartCM.hxx"
 #include "Driving.hxx"
 #include "EventHandler.hxx"
 #include "ControllerDetector.hxx"
 #include "Joystick.hxx"
 #include "Keyboard.hxx"
-#include "KidVid.hxx"
 #include "Genesis.hxx"
-#include "MindLink.hxx"
-#include "CompuMate.hxx"
-#include "AmigaMouse.hxx"
-#include "AtariMouse.hxx"
-#include "TrakBall.hxx"
 #include "Lightgun.hxx"
 #include "QuadTari.hxx"
 #include "Joy2BPlus.hxx"
@@ -899,27 +892,6 @@ void Console::createAudioQueue()
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 void Console::setControllers(string_view romMd5)
 {
-  // Check for CompuMate scheme; it is special in that a handler creates both
-  // controllers for us, and associates them with the bankswitching class
-  if(myCart->detectedType() == "CM")
-  {
-    myCMHandler = make_shared<CompuMate>(*this, myEvent, *mySystem);
-
-    // A somewhat ugly bit of code that casts to CartridgeCM to
-    // add the CompuMate, and then back again for the actual
-    // Cartridge
-    unique_ptr<CartridgeCM> cartcm(static_cast<CartridgeCM*>(myCart.release()));
-    cartcm->setCompuMate(myCMHandler);
-    myCart = std::move(cartcm);
-
-    myLeftControl  = std::move(myCMHandler->leftController());
-    myRightControl = std::move(myCMHandler->rightController());
-    myOSystem.eventHandler().defineKeyControllerMappings(
-        Controller::Type::CompuMate, Controller::Jack::Left, myProperties);
-    myOSystem.eventHandler().defineJoyControllerMappings(
-        Controller::Type::CompuMate, Controller::Jack::Left, myProperties);
-  }
-  else
   {
     // Setup the controllers based on properties
     Controller::Type leftType =
@@ -1050,18 +1022,6 @@ unique_ptr<Controller> Console::getControllerPort(
                                         swapPaddles, swapAxis, swapDir);
       break;
     }
-    case Controller::Type::AmigaMouse:
-      controller = make_unique<AmigaMouse>(port, myEvent, *mySystem);
-      break;
-
-    case Controller::Type::AtariMouse:
-      controller = make_unique<AtariMouse>(port, myEvent, *mySystem);
-      break;
-
-    case Controller::Type::TrakBall:
-      controller = make_unique<TrakBall>(port, myEvent, *mySystem);
-      break;
-
     case Controller::Type::AtariVox:
     {
       FSNode nvramfile = myOSystem.nvramDir();
@@ -1095,23 +1055,6 @@ unique_ptr<Controller> Console::getControllerPort(
       controller = make_unique<Genesis>(port, myEvent, *mySystem);
       break;
 
-    case Controller::Type::KidVid:
-    {
-      const Controller::onMessageCallbackForced callback = [&os = myOSystem]
-      (string_view msg, bool force) {
-        const bool devSettings = os.settings().getBool("dev.settings");
-        if(force || os.settings().getBool(devSettings ? "dev.extaccess" : "plr.extaccess"))
-          os.frameBuffer().showTextMessage(msg);
-        };
-      controller = make_unique<KidVid>
-        (port, myEvent, myOSystem, *mySystem, romMd5, callback);
-      break;
-    }
-
-    case Controller::Type::MindLink:
-      controller = make_unique<MindLink>(port, myEvent, *mySystem);
-      break;
-
     case Controller::Type::Lightgun:
       controller = make_unique<Lightgun>
         (port, myEvent, *mySystem, romMd5, myOSystem.frameBuffer());
diff --git a/src/emucore/Console.hxx b/src/emucore/Console.hxx
index 510b6a1..f242cc1 100644
--- a/src/emucore/Console.hxx
+++ b/src/emucore/Console.hxx
@@ -25,7 +25,6 @@ class TIA;
 class M6502;
 class M6532;
 class Cartridge;
-class CompuMate;
 class Debugger;
 class AudioQueue;
 class AudioSettings;
@@ -481,8 +480,6 @@ class Console : public Serializable, public ConsoleIO
     // Pointer to handler for switching developer settings sets
     unique_ptr<DevSettingsHandler> myDevSettingsHandler;
 
-    // Pointer to CompuMate handler (only used in CompuMate ROMs)
-    shared_ptr<CompuMate> myCMHandler;
 
     // The currently defined display format (NTSC/PAL/SECAM)
     string myDisplayFormat;
diff --git a/src/emucore/EventHandler.cxx b/src/emucore/EventHandler.cxx
index 2a95510..21c54aa 100644
--- a/src/emucore/EventHandler.cxx
+++ b/src/emucore/EventHandler.cxx
@@ -26,7 +26,6 @@
 #include "OSystem.hxx"
 #include "Paddles.hxx"
 #include "Lightgun.hxx"
-#include "PointingDevice.hxx"
 #include "Driving.hxx"
 #include "PropsSet.hxx"
 #include "Settings.hxx"
@@ -107,7 +106,6 @@ void EventHandler::initialize()
   Paddles::setDejitterDiff(myOSystem.settings().getInt("dejitter.diff"));
   Paddles::setDigitalSensitivity(myOSystem.settings().getInt("dsense"));
   Controller::setMouseSensitivity(myOSystem.settings().getInt("msense"));
-  PointingDevice::setSensitivity(myOSystem.settings().getInt("tsense"));
   Driving::setSensitivity(myOSystem.settings().getInt("dcsense"));
   Controller::setAutoFire(myOSystem.settings().getBool("autofire"));
   Controller::setAutoFireRate(myOSystem.settings().getInt("autofirerate"));
diff --git a/src/emucore/Joystick.cxx b/src/emucore/Joystick.cxx
index 3c2fdcf..03fdfbe 100644
--- a/src/emucore/Joystick.cxx
+++ b/src/emucore/Joystick.cxx
@@ -111,33 +111,6 @@ void Joystick::updateDigitalAxes()
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 void Joystick::updateMouseAxes()
 {
-  // Mouse motion and button events
-  if(myControlID > -1)
-  {
-    // The following code was taken from z26
-    static constexpr int MJ_Threshold = 2;
-    const int mousex = myEvent.get(Event::MouseAxisXMove),
-              mousey = myEvent.get(Event::MouseAxisYMove);
-
-    if(mousex || mousey)
-    {
-      if((!(abs(mousey) > abs(mousex) << 1)) && (abs(mousex) >= MJ_Threshold))
-      {
-        if(mousex < 0)
-          setPin(DigitalPin::Three, false);
-        else if(mousex > 0)
-          setPin(DigitalPin::Four, false);
-      }
-
-      if((!(abs(mousex) > abs(mousey) << 1)) && (abs(mousey) >= MJ_Threshold))
-      {
-        if(mousey < 0)
-          setPin(DigitalPin::One, false);
-        else if(mousey > 0)
-          setPin(DigitalPin::Two, false);
-      }
-    }
-  }
 }
 
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
diff --git a/src/os/libretro/Makefile.common b/src/os/libretro/Makefile.common
index eb058e8..5d23e90 100644
--- a/src/os/libretro/Makefile.common
+++ b/src/os/libretro/Makefile.common
@@ -61,7 +61,6 @@ SOURCES_CXX := \
 	$(CORE_DIR)/emucore/CartBFSC.cxx \
 	$(CORE_DIR)/emucore/CartBUS.cxx \
 	$(CORE_DIR)/emucore/CartCDF.cxx \
-	$(CORE_DIR)/emucore/CartCM.cxx \
 	$(CORE_DIR)/emucore/CartCTY.cxx \
 	$(CORE_DIR)/emucore/CartCV.cxx \
 	$(CORE_DIR)/emucore/CartDF.cxx \
@@ -94,7 +93,6 @@ SOURCES_CXX := \
 	$(CORE_DIR)/emucore/CartWD.cxx \
 	$(CORE_DIR)/emucore/CartWF8.cxx \
 	$(CORE_DIR)/emucore/CartX07.cxx \
-	$(CORE_DIR)/emucore/CompuMate.cxx \
 	$(CORE_DIR)/emucore/Console.cxx \
 	$(CORE_DIR)/emucore/Control.cxx \
 	$(CORE_DIR)/emucore/ControllerDetector.cxx \
@@ -112,17 +110,14 @@ SOURCES_CXX := \
 	$(CORE_DIR)/emucore/Joy2BPlus.cxx \
 	$(CORE_DIR)/emucore/Joystick.cxx \
 	$(CORE_DIR)/emucore/Keyboard.cxx \
-	$(CORE_DIR)/emucore/KidVid.cxx \
 	$(CORE_DIR)/emucore/Lightgun.cxx \
 	$(CORE_DIR)/emucore/M6502.cxx \
 	$(CORE_DIR)/emucore/M6532.cxx \
 	$(CORE_DIR)/emucore/MD5.cxx \
-	$(CORE_DIR)/emucore/MindLink.cxx \
 	$(CORE_DIR)/emucore/MT24LC256.cxx \
 	$(CORE_DIR)/emucore/OSystem.cxx \
 	$(CORE_DIR)/emucore/Paddles.cxx \
 	$(CORE_DIR)/emucore/PlusROM.cxx \
-	$(CORE_DIR)/emucore/PointingDevice.cxx \
 	$(CORE_DIR)/emucore/Props.cxx \
 	$(CORE_DIR)/emucore/PropsSet.cxx \
 	$(CORE_DIR)/emucore/QuadTari.cxx \
diff --git a/src/os/libretro/libretro.cxx b/src/os/libretro/libretro.cxx
index 13e5fc0..a26e38e 100644
--- a/src/os/libretro/libretro.cxx
+++ b/src/os/libretro/libretro.cxx
@@ -507,7 +507,7 @@ void retro_set_environment(retro_environment_t cb)
   static struct retro_variable variables[] = {
     // Adding more variables and rearranging them is safe.
     { "stella_console", "Console display; auto|ntsc|pal|secam|ntsc50|pal60|secam60" },
-    { "stella_palette", "Palette colors; standard|z26|user|custom" },
+    { "stella_palette", "Palette colors; standard|user|custom" },
     { "stella_filter", "TV effects; disabled|composite|s-video|rgb|badly adjusted" },
     { "stella_ntsc_aspect", "NTSC aspect %; par|100|101|102|103|104|105|106|107|108|109|110|111|112|113|114|115|116|117|118|119|120|121|122|123|124|125|50|75|76|77|78|79|80|81|82|83|84|85|86|87|88|89|90|91|92|93|94|95|96|97|98|99" },
     { "stella_pal_aspect", "PAL aspect %; par|100|101|102|103|104|105|106|107|108|109|110|111|112|113|114|115|116|117|118|119|120|121|122|123|124|125|50|75|76|77|78|79|80|81|82|83|84|85|86|87|88|89|90|91|92|93|94|95|96|97|98|99" },
